### PR TITLE
[ISSUE-67869312] add TLS support

### PR DESCRIPTION
**Pull request by haomingzhang**

### CODE REVIEW

**Comment by chebrolus** [comment]
Can you rebase off of master and push update?

**Comment by haomingzhang** [comment]
@chebrolus rebased, thanks

File: .travis.yml [hunk 1]
```diff
@@ -12,4 +12,6 @@ install:
   - glide install
 
 script:
+  - diff -u <(echo -n) <(gofmt -d $(git ls-files | grep '.go$' | grep -v vendor))
```
Comment (chebrolus) [inline .travis.yml:15]: Hmm, something wrong with rebase. This change already on `master`
still show as diff on this PR.

Comment (haomingzhang) [reply .travis.yml:15]: @chebrolus No worry.
After the rebase, your change is also in the commit list of this PR. 
Lets select rebase&merge option, then we should be good.

**Review by chebrolus — APPROVED** [review]

**Pull request merged**